test-case: manual: add regression test case for linux#1762 (#131)
thesofproject/sof-test: test-case/manual/regresstion-linux1762-system-hang-when-sof-probe-fail.md

# Linux#1762 Regression test

> Ensure system not hang when sof probe failed

Reviewer: ```suggestion
The missing SOF firmware and tplg tests are regression tests based on a sof_probe function failure documented in [linux#1762](https://github.com/thesofproject/linux/issues/1762).
```

# Preconditions
PulseAudio should be enabled

# Test Description

> * Remove fw file to let sof probe fail

Reviewer: ```suggestion
* Check that system does not hang when expected SOF fw and tplg are missing and letting sof_probe functions fail. Once they are restored, check to make sure all is working properly.
```

## Test Case:
1. Test check-fail-no-fw.md
2. Test check-fail-no-tplg.md

> 3. "sudo rmmod snd_sof_pci" -> Ensure system not hang

Reviewer: ```suggestion
3. `sudo rmmod snd_sof_pci` -> System should not hang
```

> 4. "sudo rtcwake -m mem -s 10" -> Ensure system not hang.

Reviewer: ```suggestion
4. `sudo rtcwake -m mem -s 10` -> System should not hang
```

## Expected results

> * System should not hang when sof probe fail.

Reviewer: ```suggestion
* System should not hang when expected SOF fw and tplg are missing
```

# Linux#1762 Regression test
Ensure system not hang when sof probe failed

# Preconditions
PulseAudio should be enabled

# Test Description

> * Remove fw file to let sof probe fail

Reviewer 2: I think you also remove tplg to make tplg load fail?

## Test Case:
1. Remove fw file, via "sudo mv /lib/firmware/intel/sof /lib/firmware/intel/sof_backup"

> 2. Reboot system  -> Ensure system not hang after bootup

Reviewer 3: Two spaces before "->" , but one space after "->", please remove one before "->"

> 3. Let system enter suspend & wakeup via "sudo rtcwake -m mem -s 10" ->Ensure system not hang

Reviewer 3: there should be a SPACE after "->"

> 4. Recover fw file via "sudo mv /lib/firmware/intel/sof_backup /lib/firmware/intel/sof"

Reviewer 3: Here use Restore, not Recover

5. Remove fw-tplg file, via "sudo mv /lib/firmware/intel/sof-tplg /lib/firm/intel/sof-tplg_backup"
6. Reboot system -> Ensure sytem not hang
7. "sudo rtcwake -m mem -s 10" -> Ensure system not hang

> 8. Recover fw-tplg file 

Reviewer 3: Recover -> Restore

9. Reboot system -> sof is loaded sucessfully.
10. Remove fw file
11. "sudo rtcwake -m mem -s 10" -> Ensure system not hang.

> 12. Recover fw file and reboot -> sof is loaded sucessfully

Reviewer 3: Recover -> Restore

> 13. "sudo rmmmod snd_sof_pci" -> Ensure system not hang

Reviewer 3: rmmmod -> rmmod
